### PR TITLE
Enable intuitive logs for community challenges and reporting issues

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Background

(Code heavily inspired by Toran)
It's nice to have a log of: 
- the current context for each cycle
- the full message history for each cycle
- other llm interactions (sumamrization, self feedback, scrape data)

### Changes

The log is structured like this: 
Screenshot 2023-04-30 at 8 03 34 PM
Screenshot 2023-04-30 at 8 04 01 PM

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: autogpt/log_cycle/log_cycle_mixin.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import json
+import logging
+import os
```
Comment (ntindle) [inline autogpt/log_cycle/log_cycle_mixin.py:2]: Can this be built into the logger class from logs.py that’s already imported everywhere?

Comment (waynehamadi) [reply autogpt/log_cycle/log_cycle_mixin.py:2]: oh yeah I forgot to do that

Comment (waynehamadi) [reply autogpt/log_cycle/log_cycle_mixin.py:2]: done, now this mixin is now exclusively using logger, thanks for the catch @ntindle

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Pull request closed without merging**